proxy/src/bind.rs

```diff
@@ -214,6 +216,19 @@ where
         // TODO: Add some sort of backoff logic.
         Reconnect::new(proxy)
     }
+
+    pub fn new_binding(&self, ep: &Endpoint, protocol: &Protocol) -> Binding<B> {
+        if protocol.can_reuse_clients() {
+            Binding::Bound(self.bind_stack(ep, protocol))
+        } else {
+            Binding::BindsPerRequest {
+                endpoint: ep.clone(),
+                protocol: protocol.clone(),
+                bind: self.clone(),
+                next: None
```

Comment:
Nit: since we always want to initialize `next` to `None`, consider a constructor `Binding::per_request(&Endpoint, &Protocol, &Bind)` or something. Not important, but just expresses the intention slightly more clearly, IMO.